File: vector-arithmetic/Fortran/src/mmult3.f
        SUBROUTINE mmult (a, b, c, n)
                ! Matrix A times matrix B plus matrix C
                ! column version
                ! GVL Algorithm 1.1.7

                ! Input:         A, B, C, n x n (square) matrices
                !               n, integer, dimension of matrix
                ! Output:       AB + C, stored in C

                IMPLICIT none
                DOUBLE PRECISION a(*), b(*), c(*)
                INTEGER n

                ! Internal variables
                INTEGER j, k
                INTEGER aoff, boff, coff

                DO j = 1,n
                        DO k = 1,n
                                aoff = k
                                boff = (k - 1) * n + j
                                coff = j
                                CALL saxpy ( b(boff), a(aoff), n,
     &                                  c(coff), n, n )
                        END DO
                END DO

        END SUBROUTINE

File: vector-arithmetic/Matlab/src/mmult3.m
function D = mmult3( A, B, C )
% Matrix A times matrix B plus matrix C
% column version
% GVL Algorithm 1.1.7

% Input:    A, B, C, n x n (square) matrices
% Output:   D, an n x n matrix

n = size(A);
n = n(1);

D = C;
for j = 1:n
    for k = 1:n
        D(:,j) = saxpy(B(k,j), A(:,k), D(:,j));
    end
end
end